packages/vite/src/node/server/index.ts

```diff
@@ -489,6 +495,18 @@ export async function createServer(
     handleFileAddUnlink(normalizePath(file), server)
   })
 
+  ws.on('vite:invalidate', async (url: string) => {
+    const mod = moduleGraph.urlToModuleMap.get(url)
+    if (mod && mod.isSelfAccepting && mod.lastHMRTimestamp > 0) {
+      const importers = new Set<ModuleNode>()
+      for (const importer of mod.importers) {
+        importers.add(importer)
+      }
+      const file = getShortName(mod.file!, config.root)
+      updateModules(file, Array.from(importers), mod.lastHMRTimestamp, server)
```

Comment:
```suggestion
      const file = getShortName(mod.file!, config.root)
      updateModules(file, [...mod.importers], mod.lastHMRTimestamp, server)
```
IIUC this is just converting `Set` to `Array`.

Reply:
Yes good catch. This was an oversight as I refactored from an array of module nodes (where deduping was necessary) to a single node